alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunnerService.kt

```diff
@@ -186,20 +197,22 @@ object MonitorRunnerService : JobRunner, CoroutineScope, AbstractLifecycleCompon
     override fun doClose() { }
 
     override fun postIndex(job: ScheduledJob) {
-        if (job !is Monitor) {
-            throw IllegalArgumentException("Invalid job type")
-        }
-
-        launch {
-            try {
-                monitorCtx.moveAlertsRetryPolicy!!.retry(logger) {
-                    if (monitorCtx.alertIndices!!.isAlertInitialized(job.dataSources)) {
-                        moveAlerts(monitorCtx.client!!, job.id, job)
+        if (job is Monitor) {
+            launch {
+                try {
+                    monitorCtx.moveAlertsRetryPolicy!!.retry(logger) {
+                        if (monitorCtx.alertIndices!!.isAlertInitialized(job.dataSources)) {
+                            moveAlerts(monitorCtx.client!!, job.id, job)
+                        }
                     }
+                } catch (e: Exception) {
+                    logger.error("Failed to move active alerts for monitor [${job.id}].", e)
                 }
-            } catch (e: Exception) {
-                logger.error("Failed to move active alerts for monitor [${job.id}].", e)
             }
+        } else if (job is Workflow) {
+            // do nothing
```

Comment:
We should call `moveAlerts(monitorCtx.client!!, job.id, job)` also for any changes to the workflow.

Reply:
Also we should update `moveAlerts` to also reference workflows on top of monitors

Reply:
added